diff --git a/src/megalodon.ml b/src/megalodon.ml
index 36bfc97..aeddc7c 100644
--- a/src/megalodon.ml
+++ b/src/megalodon.ml
@@ -2361,7 +2361,7 @@ let evaluate_pftac_1 pitem thmname i gpgtm gphv pfggphv =
 		  raise (Failure("Proof doesn't prove the proposition."))
 	      | Some(dl) ->
 		  deltaset := dl;
-                  if !sexprinfo then (List.iter (fun d -> Printf.printf "(DELTA \"%s\")\n" d) dl; Printf.printf "(QED)\n");
+                  if !sexprinfo then (List.iter (fun d -> Printf.printf "(DELTA \"%s\")\n" d) dl; Printf.printf "(PROOF \"%s\" %s)\n" thmname (pf_to_sexpr dgpf); Printf.printf "(QED)\n");
                   if !pfgout then List.iter (fun d -> Hashtbl.add pfgdelta d ()) !deltaset;
 		  if (!verbosity > 19) then (Printf.printf "Delta Set:"; List.iter (fun h -> Printf.printf " %s" h) dl; Printf.printf "\n"; flush stdout);
 		  let dhv = ppf_id (i,dgpf) sigtmof sigdelta in
